### PR TITLE
fix plugin fail when pom file not available

### PR DESCRIPTION
**Pull request by birukovsergey**
this commit fix plugin crash when pom file not available in remote maven repository

### CODE REVIEW

File: src/main/groovy/com/lazan/dependency/export/MavenDependencyExport.groovy [hunk 1]
```diff
@@ -71,25 +72,28 @@ class MavenDependencyExport extends DefaultTask {
...
-			if (exportSources)
-				copyAdditionalArtifacts(config, modelResolver, SourcesArtifact)
-			if (exportJavadoc)
-				copyAdditionalArtifacts(config, modelResolver, JavadocArtifact)
-		}
-		Set<String> exportedPaths = new TreeSet()
-		project.fileTree(exportDir).visit {
-			if (!it.directory) {
-				exportedPaths << it.relativePath.pathString
+		project.getAllprojects().each {currentProject ->
```
Comment (uklance) [inline src/main/groovy/com/lazan/dependency/export/MavenDependencyExport.groovy]: Hi, thanks for your PR but I won't be able to accept this in it's current state. This plugin should not apply to allprojects {}. The build author should be able to apply the plugin to the projects they want.

**Comment by birukovsergey** [comment]
how abaut this?
default value for 'project' propery is current project
permissible values - ALL_PROJECTS - fro all projects or string array with project names

**Comment by uklance** [comment]
It's standard practice for a plugin to apply to a single project. If you want to apply the plugin to all of your projects you can simply do
```
allprojects {
   apply plugin: "com.lazan.dependency-export"
}
```

**Comment by birukovsergey** [comment]
Really ... It seems I have complicated everything too much. What do you say about unavailable pom files?

**Comment by uklance** [comment]
The default behaviour should be to fail on unresolved artifacts. I'd probably want a flag which can be set.

```
failOnUnresolvedArtifact = false 
```

**Comment by birukovsergey** [comment]
like this?

**Pull request closed without merging**